File: Queue/CircularQueue.c
#include "stdio.h"
#include "stdlib.h"

struct CircularQueue{
    int size;
    int front;
    int rear;
    int *Q;
};

void Create(struct CircularQueue *q, int size){
    q->size = size+1;
    q->front=q->rear=0;
    q->Q = (int *)malloc(q->size * sizeof(int));
}

void Display(struct CircularQueue q){
    int i = q.front;
    do{
        i = (i+1)%q.size;
        printf("%d ", q.Q[i]);
    } while (i!=q.rear);
}

void Enqueue(struct CircularQueue *q, int val){
    if ((q->rear+1)%q->size==q->front)
        printf("Queue is full!\n");
    else{
        q->rear = (q->rear+1)%q->size;
        q->Q[q->rear] = val;
    }
}

int Dequeue(struct CircularQueue *q){
    int x = -1;
    if (q->front==q->rear)
        printf("Queue is empty!\n");
    else{
        q->front = (q->front+1)%q->size;
        x = q->front;
    }
    return x;
}

int main(){
    struct CircularQueue q;
    Create(&q,5);
    Enqueue(&q,1);
    Enqueue(&q,2);
    Enqueue(&q,3);
    Enqueue(&q,4);
    Enqueue(&q,5);
    Dequeue(&q);
    Dequeue(&q);
    Enqueue(&q,6);
    Enqueue(&q,7);
    Dequeue(&q);
    Display(q);
    return 0;
}

File: Queue/CircularQueue.py

class CircularQueue:
    def __init__(self,size=10):
        self.size = size+1
        self.front = 0
        self.rear = 0
        self.Q = [0] * self.size
        
    def Display(self):
        ptr = self.front
        while(ptr!=self.rear):
            ptr = (ptr+1)%self.size
            print(self.Q[ptr],end=" ")
        print()
    
    def Enqueue(self, val):
        if (self.rear+1)%self.size==self.front:
            print("Queue is full!")
        else:
            self.rear = (self.rear+1)%self.size
            self.Q[self.rear] = val
    
    def Dequeue(self):
        x = -1
        if self.front==self.rear:
            print("Queue is empty!")
        else:
            self.front = (self.front+1)%self.size
            x = self.Q[self.front]
        return x
    
if __name__=="__main__":
    q = CircularQueue(5)
    q.Enqueue(1)
    q.Enqueue(2)
    q.Enqueue(3)
    q.Enqueue(4)
    q.Enqueue(5)
    q.Enqueue(6)
    q.Display()
    q.Dequeue()
    q.Dequeue()
    q.Display()
    q.Enqueue(6)
    q.Enqueue(7)
    q.Enqueue(8)
    q.Dequeue()
    q.Display()